### ISSUE TITLE
Backport ac5a5a9 (Add default handling for single element vectors, and split/promote vNi1 vectors.) into 14.x

### ISSUE THREAD

**Issue by quinnlp** [llvm/llvm-project#53771]
Observed a stage 2 build error on Power10 for 14.0.0-rc1 bootstrap runs.

```
Assertion failed: VT.getSizeInBits() == Operand.getValueSizeInBits() && "Cannot BITCAST between types of different sizes!", file  /home/amyk/llvm/Community/llvm-project/llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp, line 5138, llvm::SDValue llvm::SelectionDAG::getNode(unsigned int, const llvm::SDLoc &, llvm::EVT, llvm::SDValue, const llvm::SDNodeFlags)()
```
Fixed by this patch: https://reviews.llvm.org/D119521

**Comment by amy-kwan** [comment]
Thanks for opening the issue. The fix has been committed ac5a5a9cfe7c83ee5fbbc48118b4239e7e6cf6c9. 
I realized later that I should have added a "Fixes: <issue number>" to the commit. My apologies.

**Comment by tstellar** [comment]
@nemanjai What do you think about backporting this? https://github.com/llvm/llvm-project/commit/ac5a5a9cfe7c83ee5fbbc48118b4239e7e6cf6c9

**Comment by nemanjai** [comment]
Yes, please do. This is a fix for a back end crash when generating Power10 code.

**Comment by tstellar** [comment]
Merged: 1e340705f142
